ui/src/types/globalConfig/entities.ts

```diff
@@ -0,0 +1,230 @@
+    defaultValue: z.union([z.string(), z.number()]).optional(),
+    options: CommonEditableEntityOptions.optional(),
+});
+
+export const TextAreaEntity = CommonEditableEntityFields.extend({
+    type: z.literal('textarea'),
+    validators: AllValidators.optional(),
+    defaultValue: z.string().optional(),
+    options: CommonEditableEntityOptions.extend({
+        rowsMin: z.number().optional(),
+        rowsMax: z.number().optional(),
+    }).optional(),
+});
+
+const AutoCompleteFields = z.array(
+    z.union([
+        ValueLabelPair,
+        z.object({
+            label: z.string(),
+            children: z.array(ValueLabelPair),
+        }),
+    ])
+);
+
+const SelectCommonOptions = CommonEditableEntityOptions.extend({
+    disableSearch: z.boolean().optional().default(false),
+    createSearchChoice: z.boolean().optional().default(false),
+    referenceName: z.string().optional(),
+    endpointUrl: z.string().optional(),
+    allowList: z.string().optional(),
+    denyList: z.string().optional(),
+    labelField: z.string().optional(),
+    autoCompleteFields: AutoCompleteFields.optional(),
+    dependencies: z.array(z.string()).optional(),
+    items: ValueLabelPair.array().optional(),
+});
+export const SingleSelectEntity = CommonEditableEntityFields.extend({
+    type: z.literal('singleSelect'),
+    validators: AllValidators.optional(),
+    defaultValue: z.union([z.string(), z.number()]).optional(),
+    options: SelectCommonOptions,
+});
+
+export const MultipleSelectEntity = CommonEditableEntityFields.extend({
+    type: z.literal('multipleSelect'),
+    validators: AllValidators.optional(),
+    defaultValue: z.string(),
+    options: SelectCommonOptions.extend({
+        delimiter: z.string().length(1),
+    }),
+});
+
+export const CheckboxEntity = CommonEditableEntityFields.extend({
+    type: z.literal('checkbox'),
+    defaultValue: z.union([z.number(), z.boolean()]),
+    options: CommonEditableEntityOptions.omit({ placeholder: true }),
+});
+
+export const CheckboxGroupEntity = CommonEditableEntityFields.extend({
+    type: z.literal('checkboxGroup'),
+    validators: z.tuple([RegexValidator]).optional(),
+    defaultValue: z.union([z.number(), z.boolean()]).optional(),
+    options: CommonEditableEntityOptions.omit({ placeholder: true }).extend({
+        groups: z
+            .array(
+                z.object({
+                    label: z.string(),
+                    fields: z.array(z.string()),
+                    options: z
+                        .object({
+                            isExpandable: z.boolean().optional(),
+                            expand: z.boolean().optional(),
+                        })
+                        .optional(),
+                })
+            )
+            .optional(),
+        rows: z.array(
+            z.object({
+                field: z.string(),
+                checkbox: z
+                    .object({
+                        label: z.string().optional(),
```

Comment:
Generally, I don't like the idea for limiting the string because it leads to bad habits. Now we support 30 symbols max. And we develop with assumption 30 symbols. Tomorrow there will be a business requirement to put 35 symbols, we will change the limit, but we have to check the whole app if it works fine.
Instead, we have to make unbreakable components that works fine with any valid input